api/mod/src/routers/users.py

```diff
@@ -0,0 +1,48 @@
+"""Routes for users"""
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.orm import Session
+
+from mod.src.database import db_models
+from mod.src.database.database import get_db
+from mod.src.models import user
+
+ROUTER = APIRouter(
+    prefix="/users",
+    tags=["users"],
+)
+
+
+@ROUTER.get(
+    "/{user_id}",
+    response_model=user.User,
+    responses={
+        404: {"description": "User not found"},
+    },
+    operation_id="read_user",
+)
+def read_user(user_id: int, db: Session = Depends(get_db)) -> user.User:
+    """Read a user by their ID"""
+    db_user = db_models.User.get_by_id(db, user_id=user_id)
+    if db_user is None:
+        raise HTTPException(status_code=404, detail="User not found")
+    return db_user
+
+
+@ROUTER.post(
+    "/",
+    response_model=user.User,
+    responses={
+        400: {"description": "Email or username already registered"},
+    },
```

Comment:
Just an idea: There should be some way to provide some sort of "model" for this 400 response that makes it possible to differentiate between "username already registered" and "email already registered". That should then also automatically generate appropriate types in the front-end. I just don't know how :sweat_smile:

Reply:
https://fastapi.tiangolo.com/advanced/additional-responses/
This is how we should actually handle error response documentation, but since this is an issue for almost all our routes, let's move this to a separate PR!